### PR TITLE
Handle inspection on cursor movement

### LINKED ISSUE #5899: Trigger inspection on cursor movement
We currently trigger inspection requests when the text content of an editor with a kernel attached to it changes. This misses cases where the user is navigating the cursor without making edits, or when they select a new editor (e.g., a new cell). 

It would be nice to also trigger inspection for those events. We would have to go to the inspection handler and wire up more editor signals to a new text editor here: https://github.com/jupyterlab/jupyterlab/blob/b6c72e27cb53d7b87415d06c8444043cbf665dcd/packages/inspector/src/handler.ts#L62-L71

In particular, in order to trigger inspection on cursor movement, we would listen to the `editor.model.selections.changed` signal. It may also be useful to wire up an `ActivityMonitor` to throttle the rate of network requests. An example of this is here: https://github.com/jupyterlab/jupyterlab/blob/b6c72e27cb53d7b87415d06c8444043cbf665dcd/packages/docregistry/src/mimedocument.ts#L55-L63

Marking this as a good issue for a new contributor.

### PR DESCRIPTION
**Pull request by ntdef**
Resolves #5899

### CODE REVIEW

**Comment by ntdef** [comment]
There's a lot of duplication in this PR but I wanted to put this out there to make sure I'm on the right track first before working on consolidating.

**Comment by ntdef** [comment]
A test failed in `@jupyterlab/test-outputarea` but it doesn't seem related to this PR>

File: packages/inspector/src/handler.ts [hunk 1]
```diff
@@ -60,14 +60,32 @@ export class InspectionHandler implements IDisposable, IInspector.IInspectable {
...
     let editor = (this._editor = newValue);
     if (editor) {
       // Clear the inspector in preparation for a new editor.
       this._cleared.emit(void 0);
-      editor.model.value.changed.connect(
-        this.onTextChanged,
+
+      this._selectionsMonitor = new ActivityMonitor({
+        signal: editor.model.selections.changed,
+        timeout: 1000
```
Comment (ian-r-rose) [inline packages/inspector/src/handler.ts]: My feeling is this is maybe a *bit* long to wait for the inspector to trigger. Maybe 250 or 500 ms?

File: packages/inspector/src/handler.ts [hunk 2]
```diff
@@ -60,14 +60,32 @@ export class InspectionHandler implements IDisposable, IInspector.IInspectable {
...
+
+      this._valueMonitor = new ActivityMonitor({
+        signal: editor.model.value.changed,
+        timeout: 1000
+      });
+      this._valueMonitor.activityStopped.connect(
+        this.onEditorChange,
         this
       );
     }
```
Comment (ian-r-rose) [inline packages/inspector/src/handler.ts]: It is possible to select a new cell/editor without changing the value or cursor position, in which case the inspector won't update. I think we should call `onEditorChange()` here as well to catch that case.

**Comment by ian-r-rose** [comment]
Also cc @saulshanabrook who has been doing some thinking about signal composition.

File: packages/inspector/src/handler.ts (hunk 1 above)
Comment (ntdef) [reply packages/inspector/src/handler.ts]: Yup, done in last commit.

File: packages/inspector/src/handler.ts (hunk 2 above)
Comment (ntdef) [reply packages/inspector/src/handler.ts]: Gotchu. Would that not trigger a `editor.model.selections.changed`? Is there another event that I should use?

**Comment by ntdef** [comment]
> What do you think about adding the ability for the `ActivityMonitor` to connect to more than one signal, with a single `activityStopped` emitter? In that case we could just connect both signals to the same monitor. I';m unsure whether it's a good idea, just musing...

Yeah, I agree, I think that would be tidy way to handle signals.

File: packages/inspector/src/handler.ts (hunk 2 above)
Comment (ian-r-rose) [reply packages/inspector/src/handler.ts]: If the selection of the new cell doesn't move the cursor, it won't trigger a change signal (I was able to verify by clicking on the same cursor position in a cell repeatedly). But you needn't listen to another event for that, we can just to call the `onEditorChange()` function in this setter directly.

**Comment by ian-r-rose** [comment]
> Yeah, I agree, I think that would be tidy way to handle signals.

Happy to keep this as is, however, and revisit it when we start implementing some of these signal-composing ideas down the line.

**Comment by ian-r-rose** [comment]
Looks good to me, thanks @ntdef!

**Pull request merged**